spectrocloud/kubevirt/schema/virtualmachineinstance/domain_spec.go

```diff
@@ -434,10 +434,10 @@ func flattenDomainSpec(in kubevirtapiv1.DomainSpec) []interface{} {
 	att := make(map[string]interface{})
 
 	att["resources"] = flattenResources(in.Resources)
-	if in.CPU != nil {
+	if in.CPU != nil && in.CPU.Cores != 0 {
 		att["cpu"] = flattenCPU(in.CPU)
 	}
-	if in.Memory != nil {
+	if in.Memory != nil && (in.Memory.Guest != nil || in.Memory.Hugepages != nil) {
```

Comment:
`flattenMemory` is not handling the case where memory is nil but hugepages is not.

Reply:
Will be done as a separate PR @TylerGillson as it's not even a part of schema.